### ISSUE TITLE
Compute list dispatched when freeing a GPUParticle2D

### ISSUE THREAD

**Issue by PouderouxVictor** [godotengine/godot#85505]
### Godot version

Godot v4.1.stable

### System information

Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3060 Ti (NVIDIA; 31.0.15.3623) - AMD Ryzen 5 5600X 6-Core Processor (12 Threads)

### Issue:
When calling queue_free() on a  GPUParticles2D the following error is sometime encoutered :
```
E 0:00:01:0519   compute_list_dispatch: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
  <Erreur C++>   Method/function failed.
  <Source C++>   drivers/vulkan/rendering_device_vulkan.cpp:8024 @ compute_list_dispatch()
```
From my testing the problem occur only if the following condition are met:
- The GPUParticle2d must have a Local To Scene Process Material ( or a duplicated Material)
- The particleShader need at least 1 uniform
- The rendering backend must be Forward+
- Several instances of the GPUParticles2D need to be present at the same time for the problem to occur

### Steps to reproduce

The reproduction project instanciate a lot of  GPUParticles2D to trigger the error fast.
Simply run the reproduction project and a lot of error will appear.

### Minimal reproduction project

[Reproduction.zip](https://github.com/godotengine/godot/files/13500974/Reproduction.zip)

**Comment by AThousandShips** [comment]
Can you please try this with a supported version, like 4.1.3 (only the latest patch version is supported, and it might already be fixed)

**Comment by AThousandShips** [comment]
Spunds like a duplicate of:
* https://github.com/godotengine/godot/issues/82257

Or similar, should be fixed in 4.1.3, please confirm

**Comment by PouderouxVictor** [comment]
Indeed it has been fixed in 4.1.3 . Thank you and sorry , I should have tested it in a more recent version .
